homeassistant/components/siren/__init__.py

```diff
@@ -0,0 +1,220 @@
+        {vol.Required(ATTR_TONE): vol.Any(vol.Coerce(int), cv.string)},
+        "async_set_default_tone",
+        [SUPPORT_TONES],
+    )
+    component.async_register_entity_service(
+        SERVICE_SET_DEFAULT_DURATION,
+        {vol.Required(ATTR_DURATION): cv.positive_float},
+        "async_set_default_duration",
+        [SUPPORT_DURATION],
+    )
+    component.async_register_entity_service(
+        SERVICE_SET_VOLUME_LEVEL,
+        {vol.Required(ATTR_VOLUME_LEVEL): cv.small_float},
+        "async_set_volume_level",
+        [SUPPORT_VOLUME_SET],
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    return await hass.data[DOMAIN].async_setup_entry(entry)
+
+
+async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    return await hass.data[DOMAIN].async_unload_entry(entry)
+
+
+class SirenEntity(ToggleEntity):
+    """Representation of a siren device."""
+
+    _attr_volume_level: float | None = None
+    _attr_default_duration: int | None = None
+    _attr_default_tone: int | str | None = None
+    _attr_available_tones: list[int] | list[str] | None = None
+
+    @property
+    def capability_attributes(self) -> dict[str, Any] | None:
+        """Return capability attributes."""
+        supported_features = self.supported_features or 0
+
+        if supported_features & SUPPORT_TONES:
+            return {ATTR_AVAILABLE_TONES: self.available_tones}
+
+        return None
+
+    @property
+    def state_attributes(self) -> dict[str, Any]:
+        """Return the optional state attributes."""
+        supported_features = self.supported_features or 0
+        data: dict[str, Any] = {}
+
+        if self.volume_level is not None:
+            data[ATTR_VOLUME_LEVEL] = self.volume_level
+
+        if self.default_duration is not None:
+            data[ATTR_DEFAULT_DURATION] = self.default_duration
+
+        if supported_features & SUPPORT_TONES:
+            data[ATTR_DEFAULT_TONE] = self.default_tone
+
+        return data
+
+    @property
+    def volume_level(self) -> float | None:
+        """Return the volume level of the device (0 - 1)."""
+        return self._attr_volume_level
+
+    @property
+    def default_duration(self) -> int | None:
+        """Return the default duration in seconds of the noise."""
+        return self._attr_default_duration
+
+    @property
+    def default_tone(self) -> int | str | None:
+        """
+        Return the default tone for the siren.
+
+        Requires SUPPORT_TONES.
+        """
+        return self._attr_default_tone
+
+    @property
+    def available_tones(self) -> list[int] | list[str] | None:
+        """
+        Return a list of available tones.
+
+        Requires SUPPORT_TONES.
+        """
+        return self._attr_available_tones
+
+    def set_default_tone(self, tone: int | str) -> None:
```

Comment:
Wonder if we should even add these separate services. We're introducing an inconsistency where an integration may support changing duration, tone, volume but may not support setting defaults.

Reply:
My original thought was that the default duration, tone, and volume could be stored in the state of the entity in cases where the device doesn't support it, but in order to do that the integration owner would have to leverage the `restore_state` helper. We could instead remove these services and let the integration decide how to handle it for cases where it's supported. We could also add more supported features flags but I'm not sure if that makes sense

Reply:
Yeah, I'm leaning to just having the `turn_on` service and use the supported features for that service, similar to how light does it. It's important that a supported feature only means one thing. There should be no options or confusion around that.

Reply:
sounds good

Reply:
should we get rid of the extra state attributes as well?